src/override-validator.ts

```diff
@@ -0,0 +1,21 @@
+import { TLRUCache } from './cache/tlru-cache';
+
+const FIVE_MUNITES_IN_MS = 5 * 3600 * 1000;
+
+export class OverrideValidator {
+  private validApiKeyCache = new TLRUCache(100, FIVE_MUNITES_IN_MS);
```

Comment:
nice thinking to cache so we're not firing off a public API request for every request